lib/split/user.rb

```diff
@@ -83,11 +83,30 @@ def alternative_key_for_experiment(experiment)
     private
 
     def keys_without_experiment(keys, experiment_key)
-      keys.reject { |k| k.match(Regexp.new("^#{experiment_key}(:finished)?$")) || k.match("#{experiment_key}:time_of_assignment") }
+      if experiment_key.include?(':')
+        sub_keys = keys.reject { |k| k.match(Regexp.new("^#{experiment_key}$")) }
+        sub_keys.reject do |k|
+          if k.match(Regexp.new("^#{experiment_key}:"))
```

Comment:
To make sure we don't include the case like `exp805_v2` as part of `exp805`

Reply:
Isn't this already filtered out of the input of `keys_without_experiment` on line 41?

Reply:
In method `max_experiments_reached?`, the usage of it is `keys_without_experiment(user.keys, experiment_key)`. The input contains other experiment keys as well.

Reply:
gotcha, I see